pkg/checks/external/main.go

```diff
@@ -376,41 +380,55 @@ func (ext *Checker) setUUID(uuid string) error {
-
-	// update the resource with the new values we want
-	ext.log("Updating khstate", checkState.Name, checkState.Namespace, "to setUUID:", checkState.Spec.CurrentUUID)
+	ext.log("Updating khstate to CurrentUUID:", checkState.Spec.CurrentUUID)
 	_, err = ext.KHStateClient.KuberhealthyStates(ext.CheckNamespace()).Update(&checkState)
+	if err != nil {
+		log.Errorln("failed to update khstate CurrentUUID for check", checkState.Namespace, checkState.Name, "with error:", err)
+	}
 
-	// Sometimes a race condition occurs when a pod has to verify uuid with kh server. If the pod happens to check the
-	// uuid before it shows as updated on kube-apiserver, the pod will not be allowed to report its status.
-	// This for-loop is to verify the pod uuid is properly set with the api server before the checker pod is started.
+	// Verify that the new CurrentUUID is set with the server by fetching it.  If something goes
+	// wrong, try to set it again up to maxTries times.
 	tries := 0
+	maxTries := 9
 	for {
-		if tries >= 9 {
-			return fmt.Errorf("failed to verify uuid match %s after %d tries", checkState.Spec.CurrentUUID, tries)
+		if tries >= maxTries {
+			return fmt.Errorf("failed to verify uuid %s was set for khstate %s after %d tries", checkState.Spec.CurrentUUID, checkState.Namespace+checkState.Name, tries)
 		}
 		tries++
-		ext.log("Waiting 1 second before checking " + ext.Name() + " uuid.")
-		time.Sleep(time.Second * 1)
+
+		// fetch the check we just updated and ensure it set properly
 		extCheck, err := ext.KHStateClient.KuberhealthyStates(ext.CheckNamespace()).Get(ext.Name(), metav1.GetOptions{})
 		if err != nil {
-			ext.log("failed to get khstate while truing up check uuid:", err)
+			ext.log("error: failed to get khstate while verifying check uuid:", err)
+			time.Sleep(time.Second)
 			continue
 		}
 		if checkState.Spec.CurrentUUID == extCheck.Spec.CurrentUUID {
-			ext.log(ext.Name() + " verified uuid match.")
+			ext.log(ext.Name() + " CurrentUUID update verified.")
 			break
 		}
+
+		// in this circumstance, the khstate has been fetched, but the CurrentUUID value on it is not the one we set
+		log.Warningln("during verification of the CurrentUUID being properly set on khstate", checkState.Namespace, checkState.Name, "UUID setting, we saw UUID", extCheck.Spec.CurrentUUID, "but expected UUID", checkState.Spec.CurrentUUID)
+		time.Sleep(time.Second * 1)
+
+		// Retry the fetch, CurrentUUID update, and set again
+		ext.log("Retrying khstate update to set CurrentUUID:", checkState.Spec.CurrentUUID)
+		checkState, err := ext.getKHState()
+		if err != nil {
+			log.Errorln("failed to fetch khstate for check", checkState.Namespace, checkState.Name, "with error:", err)
+		}
+		checkState.Spec.CurrentUUID = uuid
+		_, err = ext.KHStateClient.KuberhealthyStates(ext.CheckNamespace()).Update(&checkState)
+		if err != nil {
+			log.Errorln("failed to update khstate CurrentUUID for check", checkState.Namespace, checkState.Name, "with error:", err)
+		}
```

Comment:
I modify some details. 
```
checkStateUpdate, err = ext.KHStateClient.KuberhealthyStates(ext.CheckNamespace()).Update(&checkState)
	if err != nil {
		log.Errorln("failed to update khstate CurrentUUID for check", checkState.Namespace, checkState.Name, "with error:", err)
	}

	// Verify that the new CurrentUUID is set with the server by fetching it.  If something goes
	// wrong, try to set it again up to maxTries times.
	tries := 0
	maxTries := 9
	for {
		if tries >= maxTries {
			return fmt.Errorf("failed to verify uuid %s was set for khstate %s after %d tries", checkState.Spec.CurrentUUID, checkState.Namespace+checkState.Name, tries)
		}
		tries++
		if checkStateUpdate.Spec.CurrentUUID == uuid {
			ext.log(ext.Name() + " CurrentUUID update verified.")
			break
		}

		// in this circumstance, the khstate has been fetched, but the CurrentUUID value on it is not the one we set
		log.Warningln("during verification of the CurrentUUID being properly set on khstate", checkState.Namespace, checkState.Name, "UUID setting, we saw UUID", checkStateUpdate.Spec.CurrentUUID, "but expected UUID", uuid)
		time.Sleep(time.Second * 1)

		// Retry the fetch, CurrentUUID update, and set again
		ext.log("Retrying khstate update to set CurrentUUID:", uuid)
		checkStateNew, err := ext.getKHState()
		if err != nil {
			log.Errorln("failed to fetch khstate for check", checkState.Namespace, checkState.Name, "with error:", err)
		}
		checkStateNew.Spec.CurrentUUID = uuid
		checkStateUpdate, err = ext.KHStateClient.KuberhealthyStates(ext.CheckNamespace()).Update(&checkStateNew)
		if err != nil {
			log.Errorln("failed to update khstate CurrentUUID for check", checkState.Namespace, checkState.Name, "with error:", err)
		}
	}
```
Or we can use  retry func of the package to ensure it set properly
```
"k8s.io/client-go/util/retry"
func RetryOnConflict(backoff wait.Backoff, fn func() error) error {
	return OnError(backoff, errors.IsConflict, fn)
}

	updateErr := retry.RetryOnConflict(retry.DefaultBackoff, func() error {

		extCheck, err := ext.KHStateClient.KuberhealthyStates(ext.CheckNamespace()).Get(ext.Name(), metav1.GetOptions{})
		if err != nil {
			return err
		}
		if extCheck.Spec.CurrentUUID == uuid {
			return nil
		}
		extCheck.Spec.CurrentUUID = uuid
		_, err = ext.KHStateClient.KuberhealthyStates(ext.CheckNamespace()).Update(&extCheck)
		return err
	})
	return updateErr
```